typegraph/deno/src/utils/func_utils.ts

```diff
@@ -1,6 +1,63 @@
 // Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
 // SPDX-License-Identifier: MPL-2.0
 
+import { CREATE, DELETE, NONE, UPDATE } from "../effects.ts";
+import { InjectionSource, InjectionValue } from "./type_utils.ts";
+
+export function stringifySymbol(symbol: symbol) {
+  const name = symbol.toString().match(/\((.+)\)/)?.[1];
+  if (!name) {
+    throw new Error("unable to determine symbol name");
+  }
+  return name;
+}
+
+export function serializeInjection(
+  source: InjectionSource,
+  value: InjectionValue<unknown>,
+  valueMapper = (value: InjectionValue<unknown>) => value,
+) {
+  if (
+    typeof value === "object" &&
+    !Array.isArray(value) &&
+    value !== null
+  ) {
+    // Note:
+    // Symbol changes the behavior of keys, values, entries => props are skipped
+    const symbols = [UPDATE, DELETE, CREATE, NONE];
+    const noOtherType = Object.keys(value).length == 0;
+    const isPerEffect = noOtherType &&
+      symbols
+        .map((symbol) => (value as any)?.[symbol] !== undefined)
+        .reduce(
```

Comment:
[Array.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)

Reply:
Indeed this should make it shorter/cleaner.